Added retry code for i/o time out problem

mattcui opened the pull request:
@maximilien This PR is for addressing [issue 121](https://github.com/maximilien/softlayer-go/issues/121), I already did some test through `bosh-softlayer-go`. Please review the code change. Thanks.

maximilien commented on client/http_client.go:
```diff
@@ -164,11 +166,25 @@ func (slc *HttpClient) makeHttpRequest(url string, requestType string, requestBo
 
-	resp, err := slc.HTTPClient.Do(req)
+	var resp *http.Response
+	for i:=1; i<=5; i++ {
```
I don't like this hard coded retry count. Same for the actual wait time. We need these to be set as variables that can be changed by passing environment variables: `SL_API_RETRY_COUNT` and `SL_API_WAIT_TIME` if these environment variables are not set then I would be OK to defaul with 5 retries and 5 seconds wait.

maximilien commented:
@mattcui please address comment and please add unit tests verifying retries and environment variables to tune behavior. Thx

mattcui commented:
@maximilien Addressed your comments about setting count and wait time in environment variables. I found there is empty in [http_client_test.go](https://github.com/mattcui/softlayer-go/blob/master/client/http_client_test.go), I would like to help write unit tests there, but I hope to complete some other urgent  eCPI/Softlayer-go bugs/features first. Is it OK? Thanks.

mattcui commented:
@maximilien  It's ready for review.  Thanks.

maximilien commented:
There are no tests. Rejecting. Please add some tests (unit and at least one integration) and I will review again. Thx.
